tests/wasm/entities.ts

```diff
@@ -24,18 +24,41 @@ export function toData<T extends Names>(name: T, entity: Entity): Data<T> {
         id: getValueOf(entity, 'id', ValueKind.String),
         fromBatchId: getValueOf(entity, 'fromBatchId', ValueKind.BigInt),
         createEpoch: getValueOf(entity, 'createEpoch', ValueKind.BigInt),
-        txHash: getBytes(entity, 'txHash'),
+        txHash: getValueOf(entity, 'txHash', ValueKind.Bytes),
       }
     default:
       throw new Error(`unknown entity ${name}`)
   }
 }
 
-type ValueOf<T extends ValueKind> = Extract<Value, { kind: T }>['data']
+export function fromData<T extends Names>(name: T, data: Data<T>): Entity {
+  switch (name) {
+    case 'User':
```

Comment:
I'm not sure if this is already addressed in a future PR (the data driven one), but it would be nice if all custom specification that is use case specific could be done in a single place (allowing all the other logic to be eventually packaged up and reused across graph repos).

Reply:
>I'm not sure if this is already addressed in a future PR 

It is :smile: